### PR TITLE
NV_ReadPublic test

### PR DESCRIPTION
**Pull request by whooo**
Add test for NV_ReadPublic and adds get_cdata to NV_DefineSpace.

@williamcroberts, do you one pull request per get_cdata call + test or can I add every test + get_cdata to this PR?

### CODE REVIEW

**Comment by williamcroberts** [comment]
> Add test for NV_ReadPublic and adds get_cdata to NV_DefineSpace.
> 
> @williamcroberts, do you one pull request per get_cdata call + test or can I add every test + get_cdata to this PR?

You can add whatever you want in the PR, just break it up logically with commits for each command. Ensure that each commit builds and passes the tests so git bisect works without false positives.

**Comment by williamcroberts** [comment]
> > Add test for NV_ReadPublic and adds get_cdata to NV_DefineSpace.
> > @williamcroberts, do you one pull request per get_cdata call + test or can I add every test + get_cdata to this PR?
> 
> You can add whatever you want in the PR, just break it up logically with commits for each command. Ensure that each commit builds and passes the tests so git bisect works without false positives.

git rebase -i origin/master and using the "x" selection to run the test suite after each commit is helpful

For example:
```bash
git rebase -i origin/master
pick 0d67832f5fda test: remove unused variables from test_Vendor_TCG_Test
x python3 -m pytest -n $(nproc) test/test_esapi.py test/test_types.py
pick ffc445472240 test: fix Vendor_TCG_TEST negative test
x python3 -m pytest -n $(nproc) test/test_esapi.py test/test_types.py
pick 1d4131aa6e4a ESAPI: test Certify
x python3 -m pytest -n $(nproc) test/test_esapi.py test/test_types.py
```

File: test/test_esapi.py [hunk 1]
```diff
@@ -962,13 +962,132 @@ def test_NV_Increment(self):
 
         nvhandle = self.ectx.NV_DefineSpace(ESYS_TR.RH_OWNER, b"", nvpub)
 
+        return
```
Comment (williamcroberts) [inline test/test_esapi.py:965]: Did you mean to commit this?

Comment (whooo) [reply test/test_esapi.py]: No, my bad, fixed now

**Review by williamcroberts — APPROVED** [review]

**Pull request merged**